pkg/rbac/middleware.go

```diff
@@ -0,0 +1,102 @@
+package rbac
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/stolostron/search-v2-api/pkg/config"
+	authv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+var kClientset *kubernetes.Clientset
+
+func KubeClient() {
+	config, err := config.GetClientConfig()
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+	kClientset, err = kubernetes.NewForConfig(config)
+
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+
+}
+
+//verifies token (userid) with the TokenReview:
+func Middleware() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			// Read the value of the client identifier from the request header
+			fmt.Fprintf(w, "%s %s %s \n", r.Method, r.URL, r.Proto)
+
+			//if there is cookie in header available use that else use the authorization:
+			var tokenId string
+			if len(r.Cookies()) != 0 {
+				for _, cookie := range r.Cookies() {
+					if cookie.Name == "acm-access-token-cookie" {
+						log.Println("Cookie: ", cookie.Value)
+						tokenId = cookie.Value
+					}
+				}
+			} else {
+				if len(r.Header.Get("Authorization")) != 0 {
+					log.Println(w, "Authorization is: %s", r.Header.Get("Authorization"))
+					tokenId = r.Header.Get("Authorization")
+				}
```

Comment:
- combine `else if`.
- same as previous comment about logging.
- respond with error if request doesn't contain the token.
```suggestion
			} else if len(r.Header.Get("Authorization")) != 0 {
				klog.V(5).Info("Got user token from Authorization header.")
				tokenId = r.Header.Get("Authorization")
			} else {
				http.Error(w, "Could not find a valid token.", http.StatusUnauthorized)
				return
			}
```